treeherder/client/thclient/client.py

```diff
@@ -552,6 +552,25 @@ def validate(self):
         for d in self.data:
             d.validate()
 
+    def get_chunks(self, chunk_size):
+        """
+        Return a list of new collections broken into chunks of size ``chunk_size``.
+
+        Each chunk will be a ``TreeherderCollection`` of the same
+        type as the original with a max of ``chunk_size`` count of
+        ``TreeherderData`` objects.
+
+        Each collection must then be POSTed individually.
+        """
+        artifact_collections = []
+        data_chunks = [self.data[o:o + chunk_size] for o
```

Comment:
you can also use zip (or even better [izip](https://docs.python.org/2/library/itertools.html#itertools.izip)) to create the chunks:
```python
data_chunks = izip(*[iter(self.data)]*chunk_size)
```